src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1518,36 +1682,52 @@ impl Cage {
                             }
                         }
 
+                        // For `CharDev` type inode, the reading happens from the Character Device
+                        // file, with each device type returning different results returned
+                        // from the `_read_chr_file` file.
                         Inode::CharDev(char_inode_obj) => {
+                            // reads from character devices by matching the device number (DevNo) of
+                            // the DeviceInode.
                             self._read_chr_file(&char_inode_obj, buf, count)
                         }
-
+                        // For `Socket` type inode, a panic is returned as socket type files are not
+                        // supported.
                         Inode::Socket(_) => {
                             panic!("pread(): Socket inode found on a filedesc fd")
                         }
-
+                        // For `Dir` type inode, an error is returned as reading from a directory is
+                        // not allowed
                         Inode::Dir(_) => syscall_error(
                             Errno::EISDIR,
                             "pread",
                             "attempted to read from a directory",
                         ),
                     }
                 }
+                // Return an error for Sockets, as they do not support the concept of seeking to a
+                // specific offset because data arrives in a continuous stream from the network.
                 Socket(_) => syscall_error(
```

Comment:
I am a bit confused about the structure of these match statements - we initially get the fd object and check if it points to a FIle, a Socket or a Pipe - but within the File matching branch we again check if the inode points to a socket - are there cases where an fd matches to a type of File but the inode matches to the socket type?

Reply:
Yes, the initial match block checks for the different types of File Descriptors (Sockets, Streams, File, Pipe, Epoll), and in the inside block of File type FileDescriptor, we check for different types of inodes(File, CharDev, Directory, Sockets).  
So, both of them are different here.